opmd_viewer/addons/pic/lpa_diagnostics.py

```diff
@@ -17,4 +20,668 @@ def __init__(self, path_to_dir):
+            'x' : [-4., 10.]   (Particles having x between -4 and 10 microns)
+            'z' : [0, 100] (Particles having x between 0 and 100 microns)
+
+        Returns
+        -------
+        A tuple with:
+        - divergence in x plane in rad
+        - divergence in y plane in rad
+        """
+        # Find the output that corresponds to the requested time/iteration
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Get particle data
+        ux, uy, uz, w = self.get_particle( var_list=['ux', 'uy', 'uz', 'w'],
+                                           t=t, iteration=iteration,
+                                           species=species )
+        # Calculate diveregence
+        div_x = wstd( ux / uz, w )
+        div_y = wstd( uy / uz, w )
+        # Return the result
+        return( div_x, div_y )
+
+    def get_emittance( self, t=None, iteration=None, species=None,
+                       select=None ):
+        """
+        Calculate the normalized RMS emittance.
+        (See K Floetmann: Some basic features of beam emittance. PRSTAB 2003)
+
+        Parameters
+        ----------
+        t : float (in seconds), optional
+            Time at which to obtain the data (if this does not correspond to
+            an available file, the last file before `t` will be used)
+            Either `t` or `iteration` should be given by the user.
+
+        iteration : int
+            The iteration at which to obtain the data
+            Either `t` or `iteration` should be given by the user.
+
+        species : string
+            Particle species to use for calculations
+
+        select : dict, optional
+            Either None or a dictionary of rules
+            to select the particles, of the form
+            'x' : [-4., 10.]   (Particles having x between -4 and 10 microns)
+            'z' : [0, 100] (Particles having x between 0 and 100 microns)
+
+        Returns
+        -------
+        A tuple with :
+        - normalized beam emittance in the x plane (pi m rad)
+        - normalized beam get_emittance in the y plane (pi m rad)
+        """
+        # Find the output that corresponds to the requested time/iteration
+        # (Modifies self.current_i and self.current_t)
+        self._find_output( t, iteration )
+        # Get particle data
+        x, y, ux, uy, w = self.get_particle(
+                                    var_list=['x', 'y', 'ux', 'uy', 'w'],
+                                    t=t, iteration=iteration,
+                                    species=species )
+        # Calculate the necessary RMS values
+        x *= 1.e-6
+        y *= 1.e-6
+        xsq = np.average( x ** 2, weights=w )
+        ysq = np.average( y ** 2, weights=w )
+        uxsq = np.average( ux ** 2, weights=w )
+        uysq = np.average( uy ** 2, weights=w )
+        xpx = np.average( x * ux, weights=w )
+        ypy = np.average( y * uy, weights=w )
```

Comment:
This is really minor, but could you replace `xpx` and `ypy` by `xux` and `yuy` (just for consistency with uxsq) ?
This change would also have to be propagated in the following 2 lines.